open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-client/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/client/OpenLineage.java

```diff
@@ -2,102 +2,73 @@
      *
-     * @param serverName   name of the server to connect to.
-     * @param newServerURL the network address of the server running the governance engine.
+     * @param serverName            name of the server to connect to
+     * @param serverPlatformURLRoot the network address of the server running the OMAS REST servers
+     * @throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException if parameter validation fails
      */
-    public OpenLineage(String serverName,
-                       String newServerURL) {
-        this.serverName = serverName;
-        this.serverURL = newServerURL;
-        this.restTemplate = new RestTemplate();
+    public OpenLineage(String serverName, String serverPlatformURLRoot)
+            throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException {
+        super(serverName, serverPlatformURLRoot);
+    }
+
+    public OpenLineage(String serverName, String serverPlatformURLRoot, String userId, String password)
+            throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException {
+        super(serverName, serverPlatformURLRoot, userId, password);
     }
 
     /**
-     * Returns the graph that the user will initially see when querying lineage. In the future, this method will be
-     * extended to condense large paths to prevent cluttering of the users screen. The user will be able to extended
-     * the condensed path by querying a different method.
-     *
-     * @param userId calling user.
-     * @param graphName MAIN, BUFFER, MOCK, HISTORY.
-     * @param scope ULTIMATE_SOURCE, ULTIMATE_DESTINATION, GLOSSARY.
-     * @param view TABLE_VIEW, COLUMN_VIEW.
-     * @param guid The guid of the node of which the lineage is queried of.
-     * @return A subgraph containing all relevant paths, in graphSON format.
-     * @throws InvalidParameterException one of the parameters is null or invalid
+     * {@inheritDoc}
      */
-    public LineageVerticesAndEdges lineage(String userId, GraphName graphName, Scope scope, View view, String guid) throws InvalidParameterException {
+    public LineageVerticesAndEdges lineage(String userId,
+                                           GraphName graphName,
+                                           Scope scope,
+                                           View view,
+                                           String guid)
+            throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException, PropertyServerException {
         String methodName = "lineage";
-        String url = "/servers/{0}/open-metadata/open-lineage/users/{1}/lineage/sources/{2}/scopes/{3}/views/{4}/entities/{5}";
-        LineageResponse lineageResponse = getRestCall(url, LineageResponse.class, serverName, userId, graphName.getText(), scope.getValue(), view.getValue(), guid);
+        LineageResponse lineageResponse = callGetRESTCall(methodName, LineageResponse.class,
```

Comment:
Response can also be error response, not only LineageResponse. In case of  errorResponse, resposne should be parsed and exceptions thrown accordingly.